### PR TITLE
Fixes up bigger shuttle wall smoothing

### LINKED ISSUE #52809: Shuttle Wall Smoothing Is Fuck
## Round ID: 
I dunno like any round in the past few days

## Testmerges:
Doesn't matter

## Reproduction:
1. Have eyes
2. Locate shuttle
3. Use eyes on shuttle

### PR DESCRIPTION
**Pull request by nightred**
## About The Pull Request

The Syndicate Infiltrator and the Abandoned Ship had some bad wall smoothing based on how areas were compared.
Without a parent area designated two sub types areas could not smooth. (/area/shuttle/abandoned/bar, /area/shuttle/abandoned/crew)

Adding area_smoothing_parent lets areas with area_limited_icon_smoothing=TRUE can now smooth with it's parent and all sub types retaining the limited smoothing effect.

Fixes #52809 

## Why It's Good For The Game

Smoother walls are sexy.

## Changelog
:cl:
fix: Fixed some shuttle wall smoothing
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Change `area_limited_icon_smoothing` to use a type path, instead of a boolean, if you are going this way.

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -60,8 +60,8 @@
 	var/list/firealarms
 	var/firedoors_last_closed_on = 0
 
-	///Boolean to limit the areas (subtypes included) that atoms in this area can smooth with. Used for shuttles.
-	var/area_limited_icon_smoothing = FALSE
+	///Type path to limit the areas (subtypes included) that atoms in this area can smooth with. Used for shuttles.
+	var/area/area_limited_icon_smoothing = FALSE
```
Comment (Rohesie) [inline code/game/area/areas.dm:64]:
```suggestion
	var/area/area_limited_icon_smoothing = null
```

File: code/game/area/areas/shuttles.dm [hunk 2]
```diff
@@ -29,6 +29,7 @@
 /area/shuttle/syndicate
 	name = "Syndicate Infiltrator"
 	ambientsounds = HIGHSEC
+	area_limited_icon_smoothing=/area/shuttle/syndicate
```
Comment (Rohesie) [inline code/game/area/areas/shuttles.dm:32]:
```suggestion
	area_limited_icon_smoothing = /area/shuttle/syndicate
```
Use spaces for readability

File: code/game/area/areas/shuttles.dm [hunk 3]
```diff
@@ -64,6 +65,7 @@
 /area/shuttle/abandoned
 	name = "Abandoned Ship"
 	requires_power = TRUE
+	area_limited_icon_smoothing=/area/shuttle/abandoned
```
Comment (Rohesie) [inline code/game/area/areas/shuttles.dm:68]:
```suggestion
	area_limited_icon_smoothing = /area/shuttle/abandoned
```

File: code/game/area/areas.dm (hunk 1 above)
Comment (nightred) [reply code/game/area/areas.dm:64]: Ya that was a stupid mistake, fixing it.

File: code/game/area/areas/shuttles.dm (hunk 2 above)
Comment (nightred) [reply code/game/area/areas/shuttles.dm:32]: cleaning this up

File: code/game/area/areas/shuttles.dm (hunk 3 above)
Comment (nightred) [reply code/game/area/areas/shuttles.dm:68]: this one also thanks

**Review by Rohesie — APPROVED** [review]

**Pull request merged**